test/parser/stream_structure_conversion_test.exs

```diff
@@ -0,0 +1,335 @@
+
+        assert_sink_stream_format(conversion_pipeline_pid, :sink, %H264{
+          stream_structure: conversion_stream_structure
+        })
+
+        assert fixture_stream_structure == conversion_stream_structure
+
+      :avc3 ->
+        assert_sink_stream_format(conversion_pipeline_pid, :sink, %H264{
+          stream_structure: {:avc3, conversion_dcr}
+        })
+
+        %{nalu_length_size: converted_nalu_length_size} =
+          H264.Parser.DecoderConfigurationRecord.parse(conversion_dcr)
+
+        {:avc3, fixture_dcr} = fixture_stream_structure
+
+        %{spss: dcr_spss, ppss: dcr_ppss, nalu_length_size: fixture_nalu_length_size} =
+          H264.Parser.DecoderConfigurationRecord.parse(fixture_dcr)
+
+        fixture_nalus =
+          Enum.map(dcr_spss, &add_length_prefix(&1, converted_nalu_length_size)) ++
+            Enum.map(dcr_ppss, &add_length_prefix(&1, converted_nalu_length_size)) ++
+            split_aus_to_nalus(fixture_buffers, {:avc3, fixture_nalu_length_size})
+
+        converted_nalus =
+          split_aus_to_nalus(converted_buffers, {:avc3, converted_nalu_length_size})
+
+        assert MapSet.equal?(MapSet.new(fixture_nalus), MapSet.new(converted_nalus))
+    end
+
+    Pipeline.terminate(fixture_pipeline_pid)
+    Pipeline.terminate(conversion_pipeline_pid)
+  end
+
+  defp perform_test(stream_structure, alignment, parser_stream_structures, identical_order?) do
+    parsers =
+      Enum.map(parser_stream_structures, fn stream_structure ->
+        %H264.Parser{
+          output_alignment: alignment,
+          output_stream_structure: stream_structure
+        }
+      end)
+
+    case stream_structure do
+      :annexb ->
+        mode =
+          case alignment do
+            :au -> :au_aligned
+            :nalu -> :nalu_aligned
+          end
+
+        Enum.each(@annexb_fixtures, fn path ->
+          pid = make_annexb_pipeline(alignment, parsers)
+          perform_annexb_test(pid, File.read!(path), mode, identical_order?)
+        end)
+
+      avc when avc in [:avc1, :avc3] ->
+        fixtures =
+          case {alignment, avc} do
+            {:au, :avc1} -> @avc1_au_fixtures
+            {:nalu, :avc1} -> @avc1_nalu_fixtures
+            {:au, :avc3} -> @avc3_au_fixtures
+            {:nalu, :avc3} -> @avc3_nalu_fixtures
+          end
+
+        Enum.each(fixtures, fn path ->
+          pids = make_avc_pipelines(path, alignment, parsers, avc)
+          perform_avc_test(pids, avc)
+        end)
+    end
+  end
+
+  defp receive_buffer_payloads(pipeline_pid, acc \\ []) do
+    receive do
+      {Membrane.Testing.Pipeline, ^pipeline_pid,
+       {:handle_child_notification, {{:buffer, buffer}, :sink}}} ->
+        receive_buffer_payloads(pipeline_pid, acc ++ [buffer.payload])
+    after
+      0 ->
+        acc
+    end
+  end
+
+  defp make_parser_chain(parsers) do
+    parsers
+    |> Enum.with_index(fn elem, index -> {elem, String.to_atom("parser#{index}")} end)
```

Comment:
If left as strings i get
```
 ** (Membrane.ParentError) Child module ""parser0"" is neither Membrane Element nor Bin.
     Make sure that given module is the right one, implements proper behaviour
     and all needed dependencies are properly specified in the Mixfile.
```

Reply:
but I found out about anonymous children and will use them here

Reply:
Sorry, for confusion, I thought that the restriction of the child names to be `tuple() | atom()` is to be introduced in Membrane Core v1.0.0, but as far as I can see it is already present in v0.12 ;) We can still either use anonymous children (in case we don't need to refer to them) or call them with the use of tuples (e.g. `{:parser, index}`)